gradle-conjure-api/src/main/java/com/palantir/gradle/conjure/api/EndpointVersionBound.java

```diff
@@ -59,11 +62,7 @@ public void setMinVersion(String minVersion) {
     }
 
     public String getMaxVersion() {
```

Comment:
I think we either want to annotate this `@Nullable`, or return an `Optional<String>` and in the method body use `return Optional.ofNullable(maxVersion);` avoiding the (java) serialization issues around non-serializable `Optional` objects.

Slight preference to return an optional because it's more obvious to API consumers. It will technically be an API break, however beginning to return nulls from a previously non-null method is also a semantic break.

Reply:
Just for future reference, with the return type changed to Optional<String>, I added @JsonInclude(Include.NON_ABSENT) and set .registerModule(new Jdk8Module()) in ObjectMapper. Without these changes the serialization ends up with "max-version":{"empty":false,"present":true}.